internal/archive/credentials_test.go

```diff
@@ -0,0 +1,262 @@
+	summary: "Parsing",
+	credsFiles: map[string]string{
+		"50test-logins": `
+machine example.netter login bar password foo
+machine example.net login foo password bar
+
+machine example.org:90 login apt password apt
+machine	example.org:8080
+login
+example	password 	 foobar
+
+machine example.org
+login anonymous
+password pass
+
+machine example.com/foo login user1 unknown token password pass1
+machine example.com/bar password pass2 login user2
+		unknown token
+machine example.com/user login user
+machine example.netter login unused password firstentry
+machine socks5h://example.last/debian login debian password rules
+`,
+	},
+	matchTests: []matchTest{
+		{"https://example.net/foo", "", "foo", "bar"},
+		{"https://user:pass@example.net/foo", "", "user", "pass"},
+		{"https://example.org:90/foo", "", "apt", "apt"},
+		{"https://example.org:8080/foo", "", "example", "foobar"},
+		{"https://example.net:42/foo", "", "foo", "bar"},
+		{"https://example.org/foo", "", "anonymous", "pass"},
+		{"https://example.com/apt", "", "", ""},
+		{"https://example.com/foo", "", "user1", "pass1"},
+		{"https://example.com/fooo", "", "user1", "pass1"},
+		{"https://example.com/fo", "", "", ""},
+		{"https://example.com/bar", "", "user2", "pass2"},
+		{"https://example.com/user", "", "user", ""},
+		{"socks5h://example.last/debian", "", "debian", "rules"},
+		{"socks5h://example.debian/", "", "", ""},
+		{"socks5h://user:pass@example.debian/", "", "user", "pass"},
+	},
+}, {
+	summary: "Bad file: No machine",
+	credsFiles: map[string]string{
+		"50test-logins.conf": `
+foo example.org login foo1 password bar
+machin example.org login foo2 password bar
+machine2 example.org login foo3 password bar
+`,
+	},
+	matchTests: []matchTest{
+		{"https://example.org/foo", "", "", ""},
+	},
+}, {
+	summary: "Bad file: Ends machine",
+	credsFiles: map[string]string{
+		"50test-logins.conf": `
+machine example.org login foo1 password bar
+machine`,
+	},
+	matchTests: []matchTest{
+		{"https://example.org/foo", "", "foo1", "bar"},
+		{"https://example.net/foo", ".*\\breached end of file while expecting machine text\\b.*", "", ""},
+		{"https://foo:bar@example.net/foo", "", "foo", "bar"},
+	},
+}, {
+	summary: "Bad file: Ends login",
+	credsFiles: map[string]string{
+		"50test-logins.conf": `
+machine example.org login foo1 password bar
+machine example.net login
+`,
+	},
+	matchTests: []matchTest{
+		{"https://example.org/foo", "", "foo1", "bar"},
+		{"https://example.net/foo", ".*\\breached end of file while expecting username text\\b.*", "", ""},
+		{"https://foo:bar@example.net/foo", "", "foo", "bar"},
+	},
+}, {
+	summary: "Matches only HTTPS",
+	credsFiles: map[string]string{
+		"50test-logins.conf": `
+machine https.example login foo1 password bar
+machine http://http.example login foo1 password bar
+`,
+	},
+	matchTests: []matchTest{
+		{"https://https.example/foo", "", "foo1", "bar"},
+		{"http://https.example/foo", "", "", ""},
```

Comment:
Is that how the logic is specified and works elsewhere?  When unadorned it's https only?

Reply:
Yes, per https://manpages.debian.org/testing/apt/apt_auth.conf.5.en.html:

> If protocol is not specified, the entry only matches https and tor+https.